ext/ldap/ldap.c

```diff
@@ -1752,7 +1748,7 @@ static void php_ldap_do_search(INTERNAL_FUNCTION_PARAMETERS, int scope)
 cleanup:
 	if (ld) {
 		/* Restoring previous options */
-		php_set_opts(ld->link, old_ldap_sizelimit, old_ldap_timelimit, old_ldap_deref, &ldap_sizelimit, &ldap_timelimit, &ldap_deref);
+		php_set_opts(ld->link, old_ldap_sizelimit, old_ldap_timelimit, old_ldap_deref, (int *)&sizelimit, (int *)&timelimit, (int *)&deref);
```

Comment:
Oh, didn't notice this earlier. These casts won't properly work for big-endian architectures, but on the other hand, we're not using the variables afterwards, so might be okay. A follow-up PR could add support for passing `NULL` values to `php_set_opts()` (which then would not assign the out parameters).